src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7887,22 +7930,15 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
             // 2. For Index: Has an accessible indexer with a single int parameter
             //    For Range: Has an accessible Slice method that takes two int parameters
 
-            PropertySymbol lengthOrCountProperty;
-
             var lookupResult = LookupResult.GetInstance();
             var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
```

Comment:
I know you didn't add this code, but it also seems incorrect.

Reply:
I followed suit. In pattern method lookup we don't want any diagnostics. We only give "unsupported type for pattern" error.